pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -0,0 +1,151 @@
+package trustedcabundle
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+)
+
+const (
+	injectionOfCABundleAnnotatoion = "security.opendatahub.io/inject-trusted-ca-bundle"
+	CAConfigMapName                = "odh-trusted-ca-bundle"
+	CADataFieldName                = "odh-ca-bundle.crt"
+)
+
+func ShouldInjectTrustedBundle(ns client.Object) bool {
+	if !strings.HasPrefix(ns.GetName(), "openshift-") && !strings.HasPrefix(ns.GetName(), "kube-") &&
+		ns.GetName() != "default" && ns.GetName() != "openshift" && !HasCABundleAnnotationDisabled(ns) {
+		return true
+	}
+	return false
+}
+
+func HasCABundleAnnotationDisabled(ns client.Object) bool {
+	if value, found := ns.GetAnnotations()[injectionOfCABundleAnnotatoion]; found {
+		enabled, err := strconv.ParseBool(value)
+		return err == nil && !enabled
+	}
+	return false
+}
+
+func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsci.DSCInitialization) error {
+	namespaceList := &corev1.NamespaceList{}
+	err := cli.List(ctx, namespaceList)
+	if err != nil {
+		return err
+	}
+
+	for i := range namespaceList.Items {
+		ns := &namespaceList.Items[i]
+		if ShouldInjectTrustedBundle(ns) {
+			if err := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit.Spec.TrustedCABundle.CustomCABundle); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// createOdhTrustedCABundleConfigMap creates a configMap  'odh-trusted-ca-bundle' -- Certificates for the cluster
+// trusted CA Cert Bundle.
+func CreateOdhTrustedCABundleConfigMap(ctx context.Context, cli client.Client, namespace string, customCAData string) error {
```

Comment:
https://docs.openshift.com/container-platform/4.12/networking/configuring-a-custom-pki.html#certificate-injection-using-operators_configuring-a-custom-pki

the configmap, will it use the openshift certificate injection mechanism? In our corporate environments, all additonal CAs and self-signed trusted certificates are specified centrally at Openshift level and then used by odh notebook controller and in workbenches in this way:

https://github.com/opendatahub-io/kubeflow/pull/43/files#diff-447c1a1ddad4e46669c4371d0d9714dad4a3368c5ccf2292b356e3b5c7441ce1R198

Reply:
Yes that is correct. Notebook controller will also be updated to reference this configmap. 

cc: @harshad16